src/test/java/org/seqdoop/hadoop_bam/TestVCFInputFormat.java

```diff
@@ -100,6 +102,10 @@ public void setup() throws IOException, NoSuchMethodException, IllegalAccessExce
             BGZFCodec.class.getCanonicalName());
         conf.setInt(FileInputFormat.SPLIT_MAXSIZE, 100 * 1024); // 100K
 
+        if (filename == "small_snpeff.vcf") {
```

Comment:
Thanks for adding the test case to cover this - it will help keep test coverage up as functionality is added. However, I think its confusing to embed exceptional code paths throughout the file like this. It shouldn't be that hard to add a separate test case for this change; but at the very least I'd request that you include comments describing why this code is here. And, since presumably the test file is invalid (white space ?), it would be useful to name it something that indicates that. Absent those things, I don't think anyone coming along after the fact will be able make any sense out of this.

Reply:
I would avoid using reference equality operators on strings - it happens to work in this case due to string folding, but these comparisons should really use .equals instead of == or !=.

Reply:
@cmnbroad @fnothaft @heuermh I've made the test standalone to avoid these issues. Hope that's acceptable to everyone!